packages/transaction-controller/src/TransactionController.ts

```diff
@@ -338,17 +338,22 @@ export class TransactionController extends BaseController<
    * @param opts - Additional options to control how the transaction is added.
    * @param opts.deviceConfirmedOn - An enum to indicate what device confirmed the transaction.
    * @param opts.origin - The origin of the transaction request, such as a dApp hostname.
+   * @param opts.requireApproval - Whether to skip creating approval for transactions.
    * @returns Object containing a promise resolving to the transaction hash if approved.
    */
   async addTransaction(
     transaction: Transaction,
     {
       deviceConfirmedOn,
       origin,
+      requireApproval,
     }: {
       deviceConfirmedOn?: WalletDevice;
       origin?: string;
-    } = {},
+      requireApproval?: boolean;
+    } = {
+      requireApproval: true,
+    },
```

Comment:
It looks like you are specifying a default for the whole argument. As you have it, if e.g. `deviceConfirmedOn` were specified, then `requireApproval` would be false, not true. Is this the intended behavior? Otherwise it might be more useful to say:

``` suggestion
    {
      deviceConfirmedOn,
      origin,
      requireApproval = true,
    }: {
      deviceConfirmedOn?: WalletDevice;
      origin?: string;
      requireApproval?: boolean;
    } = {},
```

Reply:
Good catch. Your suggestion would prevent some future issues, thanks 👍

Reply:
Auto commit suggestion not worked, doing it manually